proto/api/v2/compliance_scan_schedule_service.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+option go_package = "v2";
+option java_package = "io.stackrox.proto.api.v2";
+
+import "google/protobuf/timestamp.proto";
+import weak "google/api/annotations.proto";
+import "api/v2/common.proto";
+import "api/v2/user.proto";
+
+package v2;
+
+// Represents the role of the node within the cluster
+enum NodeRole {
+    INFRA = 0;
+    WORKER = 1;
+    MASTER = 2;
+}
+
+// ClusterScanStatus holds status based on cluster in the event that a schedule was
+// successfully applied to some clusters but not others.
+// Next available tag: 3
+message ClusterScanStatus {
+    string                    cluster_id = 1;
+    repeated string           errors     = 2;
+}
+
+// Next available tag: 7
+message BaseComplianceScanScheduleSettings {
+    bool                  auto_apply_remediations  = 1;
+    bool                  auto_update_remediations = 2;
+    bool                  one_time_scan            = 3;
+    repeated string       profiles                 = 4;
+    Schedule              schedule                 = 5;
+    repeated NodeRole     node_roles               = 6;
+}
+
+// Next available tag: 4
+message ComplianceScanScheduleConfiguration {
+    string                                scan_name            = 1;
+    BaseComplianceScanScheduleSettings    scan_schedule_config = 2;
+    repeated string                       clusters             = 3;
+}
+
+// Next available tag: 7
+message ComplianceScanScheduleStatus {
+    string                                scan_name            = 1;
+    BaseComplianceScanScheduleSettings    scan_schedule_config = 2;
+    repeated ClusterScanStatus            clusters             = 3;
+    google.protobuf.Timestamp             created_time         = 4;
+    google.protobuf.Timestamp             last_updated_time    = 5;
+    // Most recent user to update the scan settings
+    SlimUser                              created_by           = 6;
+}
+
+// Next available tag: 2
+message ComplianceScanScheduleConfigRequest {
+    string scan_name = 1;
+}
+
+// Next available tag: 2
+message ListComplianceScanScheduleResponse {
+    repeated ComplianceScanScheduleStatus schedules = 1;
+}
+
+service ComplianceScanScheduleService {
+    // ListComplianceScanSchedules lists all the compliance operator schedules for the secured clusters
+    rpc ListComplianceScanSchedules (Empty) returns (ListComplianceScanScheduleResponse) {
```

Comment:
filters?

Reply:
my thought was we would add those things later.  Figured mostly you would get the scan configs or get them by cluster.  I suppose there could be use cases to get all the configurations with a profile or on a Monday but those use cases weren't immediately evident.

Reply:
Which API will cater to the table as shown [here](https://docs.google.com/presentation/d/13kqxInksYGwR_vSfASHCa38DpAvA6ucKdKxpZk6VnSE/edit#slide=id.g15b0f56b8c3_0_173) that allows users to filter based on name and cluster.

Reply:
@md2119 that is an interesting question.  That table is basically a join between the scan configuration and the results.  This API definitely needs a get by query.  The question though becomes does this API include the ability to get aggregated results or does the results API pull scan information or some other API.  I'm going to think about that a little bit.

Reply:
@md2119 I went through the slide deck again.  The target of this API was for slides 16 on down dealing with the scheduling.  IMO, the slide you point out would fit better with the results API.  I will make sure to include something there that will provide the data in the view you mention.  I also don't see thing in the slide deck that would require query search of just the scan configs, so I will probably leave that out until we find we need it.

Let me know if you disagree.

Reply:
I copied the snippet from ACS API guidelines that is in the works below:
```
All APIs that return a collection of resource types must be paginated. Therefore, such APIs must accept pagination parameters.
```

At the very least, you must support pagination on a list endpoint.

Reply:
got it, that makes sense.